### PR TITLE
Compression bench

### PR DESCRIPTION
**Pull request by StoyanTinchev**
# Description
- Update compressor and decompressor benchmarks to use Google benchmark
- CI export to build benchmarks
- Change of compression file structure
- Added example usage of json parser library that later will be used for requests to our backend server

# Test suite
 The previous approach with cppbenchmark is replaced with google benchmark library. The tests are separated into 3 files which are associated respectively to compression tests, decompression tests and both compression and decompression depending tests.

# Environment
- Host machine -> intel core i7-7700HQ 2.80GHz * 8  with  16gb ram
- OS: pop-os
- compiler: GNU gcc-11

--------------------

#### This PR makes changes which ...

- [ x ] are thoroughly tested
- [ x ] are well documented

### CODE REVIEW

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -10,33 +10,34 @@ jobs:
     strategy:
```
Comment (boki1) [inline .github/workflows/build-and-test.yml]: Please don't commit changes in whitespaces.

File: CMakeLists.txt [hunk 2]
```diff
@@ -55,8 +54,7 @@ target_link_libraries(LibEugeneCore
         CONAN_PKG::libnop
 		CONAN_PKG::andreasbuhr-cppcoro
         CONAN_PKG::benchmark
-        CONAN_PKG::catch2
-        CONAN_PKG::cppbenchmark)
+		CONAN_PKG::nlohmann_json)
```
Comment (boki1) [inline CMakeLists.txt]: We do not need that.

Omit the whitespace changes please.

File: src/standalone/json-parse/exmpl.json [hunk 3]
```diff
@@ -0,0 +1,14 @@
+{
```
Comment (boki1) [inline src/standalone/json-parse/exmpl.json]: We don't need that.

Please remove 'json-parse'.

File: src/standalone/json-parse/main.cpp [hunk 4]
```diff
@@ -0,0 +1,59 @@
+#include <nlohmann/json.hpp>
```
Comment (boki1) [inline src/standalone/json-parse/main.cpp]: I think this example should not be placed inside the repo main branch.

**Review by boki1 — CHANGES_REQUESTED** [review]

File: src/standalone/json-parse/main.cpp (hunk 4 above)
Comment (StoyanTinchev) [reply src/standalone/json-parse/main.cpp]: Ok, I will exlude it

File: src/standalone/json-parse/exmpl.json (hunk 3 above)
Comment (StoyanTinchev) [reply src/standalone/json-parse/exmpl.json]: Sure

File: CMakeLists.txt (hunk 2 above)
Comment (StoyanTinchev) [reply CMakeLists.txt]: We will need that

File: .github/workflows/build-and-test.yml [hunk 5]
```diff
@@ -10,33 +10,34 @@ jobs:
...
       fail-fast: true
       matrix:
-        os: [ubuntu-20.04]
```
Comment (boki1) [inline .github/workflows/build-and-test.yml]: Please drop the whitespace changes.

File: CMakeLists.txt [hunk 6]
```diff
@@ -21,21 +21,20 @@ include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
 conan_basic_setup(TARGETS)
 
 set(LibEugeneBtree_SRC
-        storage/btree/Btree.h
```
Comment (boki1) [inline CMakeLists.txt]: Please drop the whitespace changes.

File: CMakeLists.txt (hunk 2 above)
Comment (boki1) [reply CMakeLists.txt]: Sure, but it is not really related to 'Compression bench' :smile:

**Review by boki1 — CHANGES_REQUESTED** [review]

File: .github/workflows/build-and-test.yml [hunk 7]
```diff
@@ -18,25 +18,26 @@ jobs:
       - uses: actions/checkout@v2
       - name: Dependency setup
         run: |
-             sudo apt update
```
Comment (boki1) [inline .github/workflows/build-and-test.yml:21]: Please remove these.

**Review by boki1 — APPROVED** [review]

**Pull request merged**